components/box/Box.js

```diff
@@ -0,0 +1,84 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import theme from './theme.css';
+
+const spacings = [0, 1, 2, 3, 4, 5, 6, 7, 8];
+
+class Box extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    element: PropTypes.node,
+    margin: PropTypes.oneOf(spacings),
+    marginBottom: PropTypes.oneOf(spacings),
+    marginLeft: PropTypes.oneOf(spacings),
+    marginRight: PropTypes.oneOf(spacings),
+    marginTop: PropTypes.oneOf(spacings),
+    padding: PropTypes.oneOf(spacings),
+    paddingBottom: PropTypes.oneOf(spacings),
+    paddingLeft: PropTypes.oneOf(spacings),
+    paddingRight: PropTypes.oneOf(spacings),
+    paddingTop: PropTypes.oneOf(spacings),
+  };
+
+  static defaultProps = {
+    element: 'div',
+    margin: 0,
+    marginBottom: 0,
+    marginLeft: 0,
+    marginRight: 0,
+    marginTop: 0,
+    padding: 0,
+    paddingBottom: 0,
+    paddingLeft: 0,
+    paddingRight: 0,
+    paddingTop: 0,
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      element,
+      margin,
+      marginBottom,
+      marginLeft,
+      marginRight,
+      marginTop,
+      padding,
+      paddingBottom,
+      paddingLeft,
+      paddingRight,
+      paddingTop,
+      ...others
+    } = this.props;
+
+    const classNames = cx(
+      theme['box'],
+      {
+        [theme[`margin-${margin}`]]: margin > 0,
+        [theme[`margin-bottom-${marginBottom}`]]: marginBottom > 0,
+        [theme[`margin-left-${marginLeft}`]]: marginLeft > 0,
+        [theme[`margin-right-${marginRight}`]]: marginRight > 0,
+        [theme[`margin-top-${marginTop}`]]: marginTop > 0,
+        [theme[`padding-${padding}`]]: padding > 0,
+        [theme[`padding-bottom-${paddingBottom}`]]: paddingBottom > 0,
+        [theme[`padding-left-${paddingLeft}`]]: paddingLeft > 0,
+        [theme[`padding-right-${paddingRight}`]]: paddingRight > 0,
+        [theme[`padding-top-${paddingTop}`]]: paddingTop > 0,
```

Comment:
because `paddingTop > 0` is the conditional here, you can't override the padding-top with `0`. Applies to all the others as well of course.

Reply:
e.g. `padding={4} paddingTop={0}` will not work.

Reply:
I fixed @nickwaelkens first remark with the default values. Now I can something like padding={4} padding-top={0} and it works!

 :ok: by @driesd on 19/10/2017 16:05:57

Reply:
I implemented @nickwaelkens his solution (first remark) with the default values. Now you can perfectly override any margin/padding with a value of 0. So, right now, `padding={4} paddingTop={0}` works very well.

 :ok: by @driesd on 19/10/2017 16:18:37